packages/auto-consensus/src/info.ts

```diff
@@ -47,8 +49,13 @@ export const solutionRanges = async (api: Api) => {
 export const shouldAdjustSolutionRange = async (api: Api): Promise<boolean> =>
   await query<boolean>(api, 'subspace.shouldAdjustSolutionRange', [])
 
-export const segmentCommitment = async (api: Api) =>
-  await query<[StorageKey<AnyTuple>, Codec][]>(api, 'subspace.segmentCommitment', [])
+export const segmentCommitment = async (api: Api) => {
+  if ((api as ApiPromise).at) {
```

Comment:
Does this mean that if `ApiPromise.at` does not exist the query will be to current blockHeight?

Reply:
exact, because api = https://github.com/autonomys/auto-sdk/blob/74a393150225e63397eae18f9bfa6c0420839424/packages/auto-utils/src/types/wallet.ts#L47
so if .at does not exist, we are already at a specific block head, and so we can't change it with .at